packages/studio-ui/src/components/CollapsibleSidebar.tsx

```diff
@@ -0,0 +1,40 @@
+import classNames from "classnames";
+import { PropsWithChildren, useCallback, useState } from "react";
+import { ReactComponent as Hamburger } from "../icons/hamburger.svg";
+
+interface CollapsibleSidebarProps {
+  side: "right" | "left";
+}
+export default function CollapsibleSidebar({
+  side,
+  children,
+}: PropsWithChildren<CollapsibleSidebarProps>): JSX.Element {
+  const [open, setOpen] = useState(true);
+  const toggleOpen = useCallback(() => {
+    setOpen(!open);
```

Comment:
Can we use the callback for `setOpen` instead of passing in the value (e.g. something like `setOpen(isOpen => !isOpen))`? Then we would be able to remove `open` from the dependency array

Reply:
oh! forgot about this-- cool!